google/cloud/spanner_dbapi/__init__.py

```diff
@@ -47,6 +47,10 @@
 # threadsafety level.
 threadsafety = 1
 
+# Cloud Spanner sessions pool, used by
+# default for the whole DB API package
+default_pool = spanner_v1.BurstyPool()
```

Comment:
Could we make this lazy and only initialize it the first time a connection is opened? Opening a session pool can be an expensive operation (depending on what the default settings for the pool are).

Reply:
I think it is lazy (at least, it doesn't post requests to a backend). In the original `BurstyPool` class init, it only creates an empty queue of the given size. And in `bind()` method (which will be called on `Connection` init) it also just saves a pointer to the database object:
https://github.com/googleapis/python-spanner/blob/39ea9489475619469044573c346349b748e28d4b/google/cloud/spanner_v1/pool.py#L247-L260

In the base init it also doesn't do anything very meaningful, only initiates a labels dictionary:
https://github.com/googleapis/python-spanner/blob/39ea9489475619469044573c346349b748e28d4b/google/cloud/spanner_v1/pool.py#L38-L41

Only when we'll try to get a session (on a transaction begin), the pool will start to actually do something:
https://github.com/googleapis/python-spanner/blob/39ea9489475619469044573c346349b748e28d4b/google/cloud/spanner_v1/pool.py#L269-L273

Reply:
We do run the risk of the `BurstyPool` implementation changing, or some future contributor changing this to another pool type with an expensive initialization. If that happens I could see users being very surprised that just importing this module causes API calls to spanner. Considering it's an easy change to make, I agree that it's a good idea to make this lazy.